pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -224,20 +195,22 @@ func (c *Controller) deleteUnusedProvisioningRequests(ctx context.Context, owned
 	return nil
 }
 
-func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Workload, relevantChecks []string, activeOrLastPRForChecks map[string]*autoscaling.ProvisioningRequest) (*time.Duration, error) {
+func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Workload, wlInfo *workloadInfo, relevantChecks []string, activeOrLastPRForChecks map[string]*autoscaling.ProvisioningRequest) (*time.Duration, error) {
 	log := ctrl.LoggerFrom(ctx)
-	var requeAfter *time.Duration
+	var requeueAfter *time.Duration
 	for _, checkName := range relevantChecks {
 		// get the config
 		prc, err := c.helper.ConfigForAdmissionCheck(ctx, checkName)
 		if err != nil {
 			// the check is not active
 			continue
 		}
+		retryStrategy := ptr.Deref(prc.Spec.RetryStrategy, kueue.DefaultRetryStrategy)
 		if !c.reqIsNeeded(wl, prc) {
 			continue
 		}
-		ac := workload.FindAdmissionCheck(wl.Status.AdmissionChecks, checkName)
+		fmt.Printf("check states: %+v, checkName: %+v", wlInfo.checkStates, checkName)
```

Comment:
remove debug logs, but maybe replace them with the proper log at V(5) if you find this useful

Reply:
Done